frontend/src/stock/StockOverview.tsx

```diff
@@ -68,7 +89,8 @@ function StockOverview() {
                                     <td>{singleItem.pricePerKilo}</td>
                                     <td>Lorem</td>
                                     <td>Lorem</td>
-                                    <td>Lorem</td>
+                                    <td><DeleteIcon onClickAction={openDeleteModal}
+                                                    idToDelete={singleItem.id}/></td>
```

Comment:
Consider to pack the Icon and the states to the Modal, so you have an "Delete" Component with the button and the modal which shows up. Also possibly for the other modals
makes the code a bit tidier :)